MSGraphSDK/MSGraphSDKTests/MSCollectionRequestTests.m

```diff
@@ -0,0 +1,175 @@
+    NSDictionary *responseObject = @{ MSCollectionValueKey : @[], MSODataNextContext : @"foo" };
+    
+    NSHTTPURLResponse *OKresponse = [[NSHTTPURLResponse alloc] initWithURL:self.testBaseURL statusCode:MSExpectedResponseCodesOK HTTPVersion:@"foo" headerFields:nil];
+    [self dataTaskCompletionWithRequest:self.requestForMock data:[NSJSONSerialization dataWithJSONObject:responseObject options:0 error:nil] response:OKresponse error:nil];
+    
+    MSURLSessionDataTask *dataTask = [collectionRequst collectionTaskWithRequest:self.requestForMock odObjectWithDictionary:^MSObject *(NSDictionary *response) {
+        XCTAssert(YES);
+        return nil;
+        }
+     completion:^(MSCollection *response, NSError *error) {
+        XCTAssertNotNil(response);
+         XCTAssertNotNil(response.value);
+         XCTAssertEqual(response.value.count,0);
+        XCTAssertNil(error);
+    }];
+    [dataTask taskWithRequest:self.requestForMock];
+    
+}
+-(void)testMSCollectionRequestWithNilNextLink{
+    MSCollectionRequest *collectionRequst = [[MSCollectionRequest alloc] initWithURL:self.testBaseURL client:self.mockClient];
+    
+    NSDictionary *firstObject = @{@"foo" : @"Bar" };
+    NSDictionary *secondObject = @{@"baz" : @"qux" };
+    NSArray *values = @[firstObject, secondObject];
+    NSDictionary *responseObject = @{ MSCollectionValueKey : values};
+    
+    NSHTTPURLResponse *OKresponse = [[NSHTTPURLResponse alloc] initWithURL:self.testBaseURL statusCode:MSExpectedResponseCodesOK HTTPVersion:@"foo" headerFields:nil];
+    [self dataTaskCompletionWithRequest:self.requestForMock data:[NSJSONSerialization dataWithJSONObject:responseObject options:0 error:nil] response:OKresponse error:nil];
+    
+    MSURLSessionDataTask *dataTask = [collectionRequst collectionTaskWithRequest:self.requestForMock
+                                                          odObjectWithDictionary:^MSObject *(NSDictionary *response) {
+                                                              MSObject *cast = [[MSObject alloc] initWithDictionary:response];
+                                                              return cast;
+                                                          }
+                                                                      completion:^(MSCollection *response, NSError *error) {
+                                                                          XCTAssertNotNil(response);
+                                                                          XCTAssertNotNil(response.value);
+                                                                          XCTAssertNil(response.nextLink);
+                                                                          XCTAssertNil(error);
+                                                                      }];
+    [dataTask taskWithRequest:self.requestForMock];
```

Comment:
In general we need to prove that the completion blocks are being invoked - it may not make sense to add such verification logic to every test, but perhaps we could add a test to the end explicitly for that purpose.